microdnf/commands/install/install.cpp

```diff
@@ -83,24 +85,28 @@ void CmdInstall::run(Context & ctx) {
 
     std::cout << std::endl;
 
-    libdnf::rpm::PackageSet result_pset(&solv_sack);
-    libdnf::rpm::SolvQuery full_solv_query(&solv_sack);
+    libdnf::Goal goal(&ctx.base);
     for (auto & pattern : *patterns_to_install_options) {
-        libdnf::rpm::SolvQuery solv_query(full_solv_query);
-        solv_query.resolve_pkg_spec(dynamic_cast<libdnf::OptionString *>(pattern.get())->get_value(), true, true, true, true, true, {});
-        result_pset |= solv_query.get_package_set();
+        goal.add_rpm_install(dynamic_cast<libdnf::OptionString *>(pattern.get())->get_value(), {}, true, {});
     }
+    goal.resolve();
 
-    download_packages(result_pset, nullptr);
+    if (!print_goal(goal)) {
+        return;
+    }
 
-    std::vector<std::unique_ptr<RpmTransactionItem>> transaction_items;
-    auto ts = libdnf::rpm::Transaction(ctx.base);
-    for (auto package : result_pset) {
-        auto item = std::make_unique<RpmTransactionItem>(package, RpmTransactionItem::Actions::INSTALL);
-        auto item_ptr = item.get();
-        transaction_items.push_back(std::move(item));
-        ts.install(*item_ptr);
+    std::cout << "Is this ok [y/N]: ";
```

Comment:
This part should be moved to a general part of the code, but not a problem right now.

Reply:
I know. It is temporal solution. The better general shared code for this must be done in future.

Reply:
So, the better code exists. I did it some time ago - function `microdnf::userconfirm()`. I will use it.